From b047026b90209b4b348b41ae5a2e8c5539a64f2b Mon Sep 17 00:00:00 2001
From: Hyunsu Cho <phcho@nvidia.com>
Date: Thu, 21 Nov 2024 23:44:03 -0800
Subject: [PATCH] Open libtreelite.so with global visibility

---
 python/treelite/core.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/python/treelite/core.py b/python/treelite/core.py
index 0393e76..195a05d 100644
--- a/python/treelite/core.py
+++ b/python/treelite/core.py
@@ -37,7 +37,7 @@ def _load_lib():
         if os.path.isdir(lib_bin_path):
             os.add_dll_directory(lib_bin_path)
 
-    lib = ctypes.cdll.LoadLibrary(lib_path[0])
+    lib = ctypes.CDLL(lib_path[0], mode=ctypes.RTLD_GLOBAL)
     lib.TreeliteGetLastError.restype = ctypes.c_char_p
     lib.log_callback = _log_callback
     lib.warn_callback = _warn_callback
-- 
2.34.1

