From ed098db2be323ed95c6f11819d5e8ae2f451b87f Mon Sep 17 00:00:00 2001
From: Hyunsu Cho <chohyu01@cs.washington.edu>
Date: Tue, 31 Oct 2023 11:53:21 -0700
Subject: [PATCH] Use typing.Sequence

---
 python/treelite/model_builder.py | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/python/treelite/model_builder.py b/python/treelite/model_builder.py
index a97590e..2f6f24f 100644
--- a/python/treelite/model_builder.py
+++ b/python/treelite/model_builder.py
@@ -2,8 +2,7 @@
 import ctypes
 import dataclasses
 import json
-from collections.abc import Sequence
-from typing import Any, Dict, List, Optional, Tuple, Union
+from typing import Any, Dict, List, Optional, Sequence, Tuple, Union
 
 from .core import _LIB, _check_call
 from .model import Model
-- 
2.42.0

