diff --git a/CMakeLists.txt b/CMakeLists.txt
index 66931b0..25a5e35 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -43,15 +43,11 @@ if(BUILD_CPP_TEST)
 endif()
 
 add_library(treelite SHARED)
-add_library(treelite_static STATIC)
 add_library(treelite_runtime SHARED)
-add_library(treelite_runtime_static STATIC)
 target_link_libraries(treelite PRIVATE objtreelite objtreelite_common)
-target_link_libraries(treelite_static PRIVATE objtreelite objtreelite_common)
 target_link_libraries(treelite_runtime PRIVATE objtreelite_runtime objtreelite_common)
-target_link_libraries(treelite_runtime_static PRIVATE objtreelite_runtime objtreelite_common)
 
-foreach(lib treelite treelite_static treelite_runtime treelite_runtime_static)
+foreach(lib treelite treelite_runtime)
   set_output_directory(${lib} ${PROJECT_BINARY_DIR})
   target_link_libraries(${lib} INTERFACE dmlc)
 endforeach()
@@ -59,7 +55,7 @@ endforeach()
 # Export install targets
 include(GNUInstallDirs)
 include(CMakePackageConfigHelpers)
-install(TARGETS treelite treelite_static treelite_runtime treelite_runtime_static
+install(TARGETS treelite treelite_runtime
                 objtreelite objtreelite_common objtreelite_runtime dmlc fmt-header-only
     EXPORT TreeliteTargets
     ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
