diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6a49810..8a30ffa 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -51,15 +51,11 @@ if(BUILD_CPP_TEST)
 endif()
 
 add_library(treelite SHARED)
-add_library(treelite_static STATIC)
 add_library(treelite_runtime SHARED)
-add_library(treelite_runtime_static STATIC)
 target_link_libraries(treelite PRIVATE objtreelite objtreelite_common)
-target_link_libraries(treelite_static PRIVATE objtreelite objtreelite_common)
 target_link_libraries(treelite_runtime PRIVATE objtreelite_runtime objtreelite_common)
-target_link_libraries(treelite_runtime_static PRIVATE objtreelite_runtime objtreelite_common)
 
-foreach(lib treelite treelite_static treelite_runtime treelite_runtime_static)
+foreach(lib treelite treelite_runtime)
   set_output_directory(${lib} ${PROJECT_BINARY_DIR})
   target_link_libraries(${lib} INTERFACE dmlc)
 endforeach()
@@ -67,7 +63,7 @@ endforeach()
 # Export install targets
 include(GNUInstallDirs)
 include(CMakePackageConfigHelpers)
-set(INSTALL_TARGETS treelite treelite_static treelite_runtime treelite_runtime_static
+set(INSTALL_TARGETS treelite treelite_runtime
                     objtreelite objtreelite_common objtreelite_runtime dmlc rapidjson)
 if(NOT FMTLIB_FROM_SYSTEM_ROOT)
   list(APPEND INSTALL_TARGETS fmt-header-only)
